iphone/cli/hooks/frameworks.js

```diff
@@ -0,0 +1,841 @@
+			path.join(this._builder.projectDir, 'platform', 'iphone')
+		];
+		for (let module of this._builder.modules) {
+			pathsToScan.push(path.join(module.modulePath, 'platform'));
+			pathsToScan.push(path.join(module.modulePath, 'Resources'));
+		}
+
+		for (let pathToScan of pathsToScan) {
+			let scanPromise = this.scanPathForFrameworks(pathToScan);
+			scanPromises.push(scanPromise);
+		}
+
+		return Promise.all(scanPromises).then((results) => {
+			for (let result of results) {
+				if (result) {
+					foundFrameworkPaths = foundFrameworkPaths.concat(result);
+				}
+			}
+		}).then(() => foundFrameworkPaths);
+	}
+
+	/**
+	 * Scans the given path for frameworks and returns framwork info objects for
+	 * each framework found
+	 *
+	 * @param {string} frameworksPath Path to scan for frameworks
+	 * @return {Promise}
+	 */
+	scanPathForFrameworks(frameworksPath) {
+		if (!fs.existsSync(frameworksPath)) {
+			return Promise.resolve();
+		}
+
+		return new Promise((resolve, reject) => {
+			fs.readdir(frameworksPath, (err, files) => {
+				if (err) {
+					reject(err);
+				}
+
+				this._logger.trace(`Scanning ${frameworksPath.cyan} for frameworks`);
+				let foundFrameworkPaths = [];
+				for (let filename of files) {
+					let possibleFrameworkPath = path.join(frameworksPath, filename);
+					if (frameworkPattern.test(possibleFrameworkPath)) {
+						this._logger.trace(`  found ${path.relative(frameworksPath, possibleFrameworkPath)}`);
+						foundFrameworkPaths.push(possibleFrameworkPath);
+					}
+				}
+
+				resolve(foundFrameworkPaths);
+			});
+		});
+	}
+
+	/**
+	 * Adds all found framworks to the Xcode project
+	 *
+	 * @param {Object} hookData Data from the Xcode project hook
+	 * @param {Function} callback Callback function
+	 */
+	addFrameworksToXcodeProject(hookData, callback) {
+		if (this._frameworks.size === 0) {
+			return callback();
+		}
+
+		let xcodeProject = hookData.args[0];
+		let frameworkIntegrator = new FrameworkIntegrator(xcodeProject, this._builder, this._logger);
+		for (let frameworkInfo of this._frameworks.values()) {
+			this._logger.trace(`Integrating ${frameworkInfo.type} framework ${frameworkInfo.name.green} into Xcode project.`);
+			frameworkIntegrator.integrateFramework(frameworkInfo);
+		}
+
+		if (this.hasFrameworksWithFatBinary()) {
+			this._logger.trace('Framework with fat binary present, integrating script to strip invalid architectures.');
+			frameworkIntegrator.integrateStripFrameworksScript();
+		}
+
+		frameworkIntegrator.adjustRunpathSearchPath();
+
+		callback();
+	}
+
+	/**
+	 * Checks all found dyanmic frameworks if any of them include both device and
+	 * simulator architectures
+	 *
+	 * @return {Boolean} True if device and simulator archs were found, false otherwise
+	 */
+	hasFrameworksWithFatBinary() {
+		let deviceArchitectures = new Set([ 'armv7', 'arm64' ]);
+		let simulatorArchitectures = new Set([ 'i386', 'x86_64' ]);
+		let hasDeviceArchitectures = false;
```

Comment:
`hasDeviceArchitectures` and `hasSimulatorArchitectures` must be either defined inside the for loop below or reset at the beginning of the loop.

Say the first framework has a device architecture, but not a simulator architecture, then `hasDeviceArchitectures` will be `true`, but `hasSimulatorArchitectures` will be `false`. Next it would loop and go to the second framework. Say the second framework does not have a device architecture, but has a simulator architecture. Since `hasDeviceArchitectures` was not reset to `false`, it will be `true` and `hasSimulatorArchitectures` will be `true` and this function will incorrectly return `true`.